packages/core/lib/console.js

```diff
@@ -130,6 +133,14 @@ class Console extends EventEmitter {
     abstractions.forEach(abstraction => {
       contextVars[abstraction.contract_name] = abstraction;
     });
+
+    // make sure the repl gets the new contracts in its context
+    if (this.repl) {
+      this.repl.context = {
+        ...this.repl.context,
+        ...this.setContextVars(contextVars)
```

Comment:
`setContextVars` can be replaced with an inline operation

```suggestion
        ...contextVars
```

Reply:
I did this a different way to ensure preservation of any assignments/scripts that a user creates within the console.